sdk/go/common/workspace/project.json

```diff
@@ -5,176 +5,290 @@
-                    "type": "array",
-                    "items": {
-                        "$ref": "#/$defs/pluginOptions"
-                    }
-                },
-                "analyzers": {
-                    "description": "Plugins for policy analyzers.",
-                    "type": "array",
-                    "items": {
-                        "$ref": "#/$defs/pluginOptions"
-                    }
-                },
-                "languages": {
-                    "description": "Plugins for languages.",
-                    "type": "array",
-                    "items": {
-                        "$ref": "#/$defs/pluginOptions"
-                    }
-                }
+        "additionalProperties": false
+      },
+      "plugins": {
+        "description": "Override for the plugin selection. Intended for use in developing pulumi plugins.",
+        "type": "object",
+        "properties": {
+          "providers": {
+            "description": "Plugins for resource providers.",
+            "type": "array",
+            "items": {
+              "$ref": "#/$defs/pluginOptions"
+            }
+          },
+          "analyzers": {
+            "description": "Plugins for policy analyzers.",
+            "type": "array",
+            "items": {
+              "$ref": "#/$defs/pluginOptions"
             }
+          },
+          "languages": {
+            "description": "Plugins for languages.",
+            "type": "array",
+            "items": {
+              "$ref": "#/$defs/pluginOptions"
+            }
+          }
         }
+      }
     },
-    "required": ["name", "runtime"],
+    "required": [
+      "name",
+      "runtime"
+    ],
     "additionalProperties": true,
     "$defs": {
-        "pluginOptions": {
-            "title": "PluginOptions",
-            "type": "object",
+      "pluginOptions": {
+        "title": "PluginOptions",
+        "type": "object",
+        "properties": {
+          "name": {
+            "type": "string",
+            "description": "Name of the plugin"
+          },
+          "path": {
+            "type": "string",
+            "description": "Path to the plugin folder"
+          },
+          "version": {
+            "type": "string",
+            "description": "Version of the plugin, if not set, will match any version the engine requests."
+          }
+        }
+      },
+      "simpleConfigType": {
+        "title": "SimpleConfigType",
+        "enum": ["string","integer", "boolean", "array"]
+      },
+      "configItemsType": {
+          "type": "object",
+          "required": ["type"],
+          "properties": {
+            "type": {
+                "oneOf": [
+                { "$ref": "#/$defs/simpleConfigType" },
+                { "$ref": "#/$defs/configItemsType" }
+              ]
+            },
+            "items": {
+              "$ref": "#/$defs/configItemsType"
+            }
+          },
+          "if": {
```

Comment:
I wonder if a "oneOf" results in better errors here than "if" 🤔 
Not needed for this first pass but I might have a play with that idea later, so which is nicer UX.